src/canmatrix/canmatrix.py

```diff
@@ -53,15 +53,16 @@
 from past.builtins import basestring
 import copy
 
+
+@attr.s
 class BoardUnit(object):
     """
     Contains one Boardunit/ECU
     """
 
-    def __init__(self, name):
-        self.name = name.strip()
-        self.attributes = dict()
-        self.comment = None
+    name = attr.ib(type=str)
+    comment = attr.ib(type=str, default=None)
```

Comment:
Shouldn't type be `typing.Optional[str]`?  Then we get to add the `typing` backport as a dependency since it is only stdlib in Python>=3.5.

Or maybe we should drop the None option and there's always a comment, it just may be an empty string?  I dunno, but that'd be a separate change anyways

Reply:
For this concrete case: the Signal has comment with default="". Make sense to have the same in BoardUnit too. (Moreover, I posed myself a question: does it make sense to have None as default, and allow the user to set an empty string? Is there any difference between `""` and `None` from user point of view? I mean: no.)
Generaly: as far as I understand, the attr `type` is only for static analysis and they encourage to replace it by new type hinting. We can't do that due to py2 compatibility. But there is still py2 annotation style (PEP484, PEP526). 
Would you agree with this?
```suggestion
    comment = attr.ib(default="")  # type: str
```

the functions are currently annotated in comments (okay, in docstrings). Annotation attr.ibs in comments would be more ... conform. When dropping py2 support once, it could be refactored to new annotation style by a regex.

I thing we could annotate for example in Frame as follows:
```python
receiver = attr.ib(factory=list)  # type:List[str]
```
with the `typing` dependency, obviously.
I thing the type annotations are useful for users and help to understand the code faster.

Reply:
How's this.  Rather than piecemeal tweaking `None` vs. `""` here then there then somewhere else  we make a ticket.  Whoever can share opinions there and we decide what it should be everywhere and then have a PR to apply that to the entire library consistently.  This is also a distinctly breaking change so it'd probably be good to be fairly explicit about it rather than continuously evolving and breaking in different places.  (or maybe we are already more consistent than I am guessing...)

Given where we are with py2 still I think that the pep-approved comment style is fine.  But I admittedly haven't read up on it yet.

Note that even though annotations already existed, the typing module is new in 3.5 so even without py2 support we will still need the typing backport to continue 3.4 support for anything but pure types.